tests/integration/targets/meraki_mr_radio/tasks/main.yml

```diff
@@ -0,0 +1,155 @@
+# Test code for the Meraki modules
+# Copyright: (c) 2021, Tyler Christiansen (@supertylerc)
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- block:
+  - name: Create network with type wireless
+    meraki_network:
+      auth_key: "{{ auth_key }}"
+      state: present
+      org_name: "{{ test_org_name }}"
+      net_name: IntTestNetworkWireless
+      type: wireless
+      timezone: America/Chicago
+    delegate_to: localhost
+
+  - name: Add access points to network
+    meraki_device:
+      auth_key: "{{ auth_key }}"
+      state: present
+      org_name: "{{ test_org_name }}"
+      net_name: IntTestNetworkWireless
+      serial: "{{ serial_wireless }}"
+    delegate_to: localhost
+
+  - name: Create RF profile
+    meraki_mr_rf_profile:
+      auth_key: "{{ auth_key }}"
+      org_name: "{{ test_org_name }}"
+      net_name: IntTestNetworkWireless
+      state: present
+      name: Test Profile
+      band_selection_type: ap
+      client_balancing_enabled: True
+      ap_band_settings:
+          mode: dual
+          band_steering_enabled: true
+      five_ghz_settings:
+        max_power: 10
+        min_bitrate: 12
+        min_power: 8
+        rxsop: -65
+        channel_width: 20
+        valid_auto_channels:
+          - 36
+          - 40
+          - 44
+      two_four_ghz_settings:
+        max_power: 10
+        min_bitrate: 12
+        min_power: 8
+        rxsop: -65
+        ax_enabled: false
+        valid_auto_channels:
+          - 1
+    delegate_to: localhost
+    register: create
+
+  - set_fact:
+      profile_id: "{{ create.data.id }}"
+
+  - name: Configure AP radios in check mode
+    meraki_mr_radio:
+      auth_key: "{{ auth_key }}"
+      org_name: "{{ test_org_name }}"
+      net_name: IntTestNetworkWireless
+      state: present
+      serial: "{{ serial_wireless }}"
+      five_ghz_settings:
+        channel: 40
+        channel_width: 20
+        target_power: 10
+      two_four_ghz_settings:
+        channel: 1
+        target_power: 10
+      rf_profile_name: Test Profile
+    delegate_to: localhost
+    register: create_check
+    check_mode: true
+
+  - assert:
+      that:
+        - create_check.data is defined
+        - create_check is changed
+
+  - name: Query a device radio
+    meraki_mr_rf_profile:
+      auth_key: '{{ auth_key }}'
+      serial: "{{ serial_wireless }}"
```

Comment:
Does this work for you? It's complaining there's no serial parameter in `meraki_mr_rf_profile`.

Reply:
That was a copy/paste mistake.  The module name should've been `meraki_mr_radio` as the query is intended to query a specific AP's radio configuration.  Fixed in 065979c.

Reply:
The default setting is causing errors.
```
TASK [meraki_mr_radio : Query a device radio] ***************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "value of channel_width must be one of: auto, 20, 40, 80, got: None found in five_ghz_settings"}
```

Default is indeed set to `None`. Fix I think would be to add `None` to the choices list or to remove the default arg parameter altogether.

Reply:
That's been a constant struggle, since Ansible's sanity tests don't like what it perceives as incompatible types (`None` vs. `int`), but having no default means some additional logic in code to set things up.  I thought I had taken care of this once and for all in a48bcd7, but I'll revisit this again and possibly just add a bunch of `if`s to deal with it.

Reply:
I've tested this on the most recent commit (a48bcd7) and didn't run into this failing.  Can you confirm which commit you were running when you encountered that error?

```bash
$ ansible-playbook -i inventory/hosts provision-wifi.yml --diff -t test

PLAY [!localhost] ******************************************************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************************************************
ok: [ap1]

TASK [manage-devices : Query a device radio] ***************************************************************************************************************************************************************
ok: [ap1 -> localhost]

TASK [manage-devices : assert] *****************************************************************************************************************************************************************************
ok: [ap1] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP *************************************************************************************************************************************************************************************************
ap1              : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```